resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/AliasManager.java

```diff
@@ -129,32 +113,17 @@ class LocationMap {
 		/**
 		 * Map of FileStore-&gt;IResource OR FileStore-&gt;ArrayList of (IResource)
 		 */
-		private final SortedMap<IFileStore, Object> map = new TreeMap<>(IFileStore::compareTo);
+		private final SortedMap<IFileStore, ArrayList<IResource>> map = new TreeMap<>(IFileStore::compareTo);
```

Comment:
The idea behind this Object was to either use single IResource or ArrayList<IResource> to safe memory in the case that the list is only 1 entry long. if you change that you should verify that the additional memory is neglectable.

Reply:
I removed this part and reduced the change to get tests passing.